### ISSUE TITLE
Pools issue

### ISSUE THREAD

**Issue by arthurmv** [shish/shimmie2#711]
After watching some random pictures I found this.
```
Fatal error: Uncaught TypeError: Argument 1 passed to Pools::get_single_pool() must be of the type int, string given, called in W:\htdocs\shimmie\ext\pools\main.php on line 339 and defined in W:\htdocs\shimmie\ext\pools\main.php:581 Stack trace: #0 W:\htdocs\shimmie\ext\pools\main.php(339): Pools->get_single_pool('24') #1 W:\htdocs\shimmie\core\send_event.php(113): Pools->onDisplayingImage(Object(DisplayingImageEvent)) #2 W:\htdocs\shimmie\ext\view\main.php(63): send_event(Object(DisplayingImageEvent)) #3 W:\htdocs\shimmie\core\send_event.php(113): ViewImage->onPageRequest(Object(PageRequestEvent)) #4 W:\htdocs\shimmie\index.php(95): send_event(Object(PageRequestEvent)) #5 {main} thrown in W:\htdocs\shimmie\ext\pools\main.php on line 581
```
Shimmie 2.8.2

**Comment by arthurmv** [comment]
UPDATE: If I desactive pool extensions it works again...

**Comment by arthurmv** [comment]
If I desactive pool extension it works again...

**Comment by shish** [comment]
What database software are you using?

**Comment by arthurmv** [comment]
MySQL

**Comment by arthurmv** [comment]
MySQL and PHPmyAdmin for management.

**Comment by shish** [comment]
I think 2d0b107 should fix that, though I don't have a mysql install handy to confirm ^^

**Comment by shish** [comment]
Looks like the pools extension has the same issue a few times (PHP converts all database data to strings, when the code wants integers, so we need to convert back...) - a bunch of fixes in master (237f8148) though I've not tested every line of code

**Comment by arthurmv** [comment]
> I think [2d0b107](https://github.com/shish/shimmie2/commit/2d0b107adb033a287bc3972cf69c459e21d84ebe) should fix that, though I don't have a mysql install handy to confirm ^^

```
Fatal error: Uncaught TypeError: Argument 1 passed to Image::by_id() must be of the type int, string given, called in W:\htdocs\shimmie\ext\pools\main.php on line 823 and defined in W:\htdocs\shimmie\core\imageboard\image.php:100 Stack trace: #0 W:\htdocs\shimmie\ext\pools\main.php(823): Image::by_id('1910') #1 W:\htdocs\shimmie\ext\pools\main.php(206): Pools->get_posts(Object(PageRequestEvent), 29) #2 W:\htdocs\shimmie\core\send_event.php(113): Pools->onPageRequest(Object(PageRequestEvent)) #3 W:\htdocs\shimmie\index.php(95): send_event(Object(PageRequestEvent)) #4 {main} thrown in W:\htdocs\shimmie\core\imageboard\image.php on line 100
```

**Comment by arthurmv** [comment]
> Looks like the pools extension has the same issue a few times (PHP converts all database data to strings, when the code wants integers, so we need to convert back...) - a bunch of fixes in master ([237f814](https://github.com/shish/shimmie2/commit/237f8148f3a4e4de633ee7b78ce22efc12a00cf3)) though I've not tested every line of code

Fixed! Thanks again!

**Comment by shish** [comment]
Heads up that 2.8.3 contains a bunch of extra pools fixes ^^
